### PR TITLE
feat: vip areas block initial implementation

### LINKED ISSUE #4: VIP Areas block
URL where the block is visible: https://www.realmadrid.com/area-vip/partido-vip

For each VIP are we need: 
* Name of the area
* Description
* Image
* Link to the detail page

Screenshot 2023-04-21 at 17 58 26

Screenshot 2023-04-21 at 17 56 44

The VIP areas data is already available in the index query.
There is a first and simplistic version of the code required to fetch the data from the index and render it. https://github.com/hlxsites/realmadrid/tree/main/blocks/vip-areas

You would need to review the existing code, make it production ready and apply the correct styling

Sample page: https://main--realmadrid--hlxsites.hlx.page/area-vip/tests/test-vip-areas

### PR DESCRIPTION
**Pull request by vdua**
Fix #4

Test URLs:
- Before: https://main--realmadrid--hlxsites.hlx.page/area-vip/partido-vip
- After: https://vip-areas--realmadrid--hlxsites.hlx.page/area-vip/partido-vip

Pending:

1. In mobile all the cards are not shown but only 3 and rest 2 on click of a button.
2. The order of cards is random while it stays the same in the new site.

### CODE REVIEW

File: blocks/vip-areas/vip-areas.css [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+.vip-areas ul li {
+  margin-bottom: 30px;
+}
+
+.vip-areas ul .info {
+  padding: 10px 12px;
+  box-sizing: border-box;
+  transform: translateY(-20px);
+  background-color: #fff;
```
Comment (chicharr) [inline blocks/vip-areas/vip-areas.css:43]: please use the variables defined in `styles.css` 
` var(--background-color)`

File: blocks/vip-areas/vip-areas.css [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+  margin: 0 auto;
+}
+
+.vip-areas ul h3 {
+  margin: 0;
+  padding: 0;
+  font-size: 20px;
+  line-height: 1.25;
+  letter-spacing: -0.2px;
+  color: #000;
```
Comment (chicharr) [inline blocks/vip-areas/vip-areas.css:54]: please use the variables defined in styles.css
 `var(--text-color)`

File: blocks/vip-areas/vip-areas.css [hunk 3]
```diff
@@ -0,0 +1,155 @@
+main .vip-areas-container {
+  width: 100%;
+  padding: 10px;
+  margin-top: 50px;
+  margin-bottom: 10px;
+  box-sizing: border-box;
+}
+
+main .vip-areas-container .default-content-wrapper h2 {
```
Comment (chicharr) [inline blocks/vip-areas/vip-areas.css:9]: This should actually go in the main styles.css as this titles are shared for different blocks not just the vip areas.
Please not that there's another PR in review which is already tackling these titles. See https://issue-3--realmadrid--hlxsites.hlx.page/area-vip/partido-vip

Comment (chicharr) [reply blocks/vip-areas/vip-areas.css:9]: just remove it this part, I added it to the styles.css as part of issue-11

**Comment by chicharr** [comment]
One last comment:
On the mobile view the link to open the page is underlined when you hover on it. Could you please remove the text decoration? 
Screenshot 2023-04-27 at 18 38 39

File: blocks/vip-areas/vip-areas.js [hunk 4]
```diff
@@ -6,20 +6,34 @@ async function fetchVIPAreas() {
   return json.data;
 }
 
+function areaTemplate(area) {
+  return `
+    <a href="${area.path}">
+      <img src="${area.image}" alt="${area.title}"/>
```
[the remark this answers is not in the document]
Comment (vdua) [reply blocks/vip-areas/vip-areas.js]: Resolved. The only issue is that the image resolution we have is higher than the one in the real-madrid site. I did not downgrade that for now

File: blocks/vip-areas/vip-areas.js [hunk 5]
```diff
@@ -6,20 +6,34 @@ async function fetchVIPAreas() {
   return json.data;
```
[the remark this answers is not in the document]
Comment (vdua) [reply blocks/vip-areas/vip-areas.js]: when you say URL configurable. Do we want it to be govern by authors or in the code itself ?

**Comment by vdua** [comment]
> On the mobile view the link to open the page is underlined when you hover on it. Could you please remove the text decoration?


Shouldn't this be removed from the base styles.css ? There is no text-decoration applied when we hover on any URL in the page (except in the nav, but that is most probably border-bottom)

File: styles/styles.css [hunk 6]
```diff
@@ -114,7 +115,9 @@ h2 {
   padding-bottom: 5px;
   color: #333;
   border-bottom: 4px solid #0255A5;
-  font-size: var(--heading-font-size-s)
+  font-size: var(--heading-font-size-s);
+  width: var(--section-heading-width);
+  border-right: 4px solid transparent;
```
Comment (vdua) [inline styles/styles.css:120]: this will cause a conflict with #16. @chicharr After you merge that, I will resolve the conflict and update.

Comment (chicharr) [reply styles/styles.css:134]: PR 16 is merged, you can proceed

File: styles/styles.css [hunk 7]
```diff
@@ -194,5 +197,5 @@ main img {
 
 main .section > div {
   width: var(--wrapper-width);
-  margin: 0 auto;
+  margin: 0 auto;
```
Comment (chicharr) [inline styles/styles.css:215]: please, configure your IDE to remove trailing whitespaces

File: blocks/vip-areas/vip-areas.js [hunk 8]
```diff
@@ -1,39 +1,44 @@
...
+    // eslint-disable-next-line no-console
+    console.log(`unable to fetch vip areas ${e}`);
+  }
+  return [];
 }
 
-function areaTemplate(area) {
-  return `
+function areaElement(area) {
+  const picture = createOptimizedPicture(area.image, area.title, false);
```
Comment (chicharr) [inline blocks/vip-areas/vip-areas.js:20]: Not sure if needed, but you can pass to the function different breakpoints. You can check the use in hero.js. 
If we need 2 different resolutions for the image, e.g. one for > 990 and one for < 990 you can add it there.

File: blocks/vip-areas/vip-areas.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (chicharr) [reply blocks/vip-areas/vip-areas.js]: there will be a config file where admins can configure the endpoint

**Review by chicharr — APPROVED** [review]
as discussed on slack, please take a look at my last comments and feel free to merge the PR. I'm approving it

File: blocks/vip-areas/vip-areas.js [hunk 9]
```diff
@@ -1,9 +1,39 @@
...
+    return json.data;
+  } catch (e) {
+    // eslint-disable-next-line no-console
+    console.log(`unable to fetch vip areas ${e}`);
+  }
+  return [];
+}
+
+function areaElement(area) {
+  const breakpoints = [
```
Comment (vdua) [inline blocks/vip-areas/vip-areas.js:20]: The breakpoints in the original site are

| media query  | image max size |
| ------------- | ------------- |
| w < 990 | 100% |
| 990 < w < 1200  | 480px  |
| 1200 < w   | 800px  |

I am not breaking down it into three but into two, using 960px image (the one used in the original site) for w > 990 and 1200px for w < 990

**Comment by vdua** [comment]
@chicharr Merging it now with the current breakpoints. Can update them if required later on.

**Pull request merged**